src/lib.rs

```diff
@@ -27,18 +32,20 @@
 // those of the authors and should not be interpreted as representing official
 // policies, either expressed or implied, of Dmitry Vyukov.
 //
-
-#![allow(missing_docs, dead_code)]
+#![no_std]
+#![allow(missing_docs)]
 
 // http://www.1024cores.net/home/lock-free-algorithms/queues/bounded-mpmc-queue
 
 // This queue is copy pasted from old rust stdlib.
+extern crate alloc;
 
-use std::sync::Arc;
-use std::cell::UnsafeCell;
+use alloc::{sync::Arc, vec::Vec};
```

Comment:
I'd prefer that we don't use this syntax for use statements. I'd rather this be:
```
use alloc::sync::Arc;
use alloc::vec::Vec;
```

Reply:
roger that, I think rust-analyzer did that automatically. I'll change all imports back to single-liners.